variation-commons-mongodb/src/main/java/uk/ac/ebi/eva/commons/mongodb/entities/VariantSourceMongo.java

```diff
@@ -98,6 +99,27 @@
         );
     }
 
+    public VariantSourceMongo(IVariantSource variantSource) {
+        this.fileId = variantSource.getFileId();
+        this.fileName = variantSource.getFileName();
+        this.studyId = variantSource.getStudyId();
+        this.studyName = variantSource.getStudyName();
+        this.type = variantSource.getType();
+        this.aggregation = variantSource.getAggregation();
+        this.samplesPosition = new HashMap<>();
+        Map<String,Integer> samplesPosition = variantSource.getSamplesPosition();
+        if (samplesPosition != null && !samplesPosition.isEmpty()) {
+            this.samplesPosition.putAll(samplesPosition);
+        }
+        this.metadata = new HashMap<>();
+        Map<String, Object> metadata = variantSource.getMetadata();
+        if (metadata != null && !metadata.isEmpty()) {
+            this.metadata.putAll(metadata);
+        }
+        this.stats = (VariantGlobalStatsMongo)(variantSource.getStats());
```

Comment:
This cast will write the wrong field names if the instance is a regular VariantGlobalStats. create a new instance of VariantGlobalStatsMongo copying the attributes of variantSource.getStats()